src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -0,0 +1,93 @@
+                }
+
+                if (string.IsNullOrEmpty(settings.EventHubName))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventProcessorClient could not be configured. Ensure a valid EventHubName was provided in the '{configurationSectionName}' configuration section.");
+                }
+
+                options.Identifier ??= GenerateClientIdentifier(settings);
+
+                var blobClient = GetBlobContainerClient(settings, cred, configurationSectionName);
+
+                var processor = !string.IsNullOrEmpty(connectionString)
+                    ? new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, connectionString)
+                    : new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, settings.Namespace,
+                        settings.EventHubName, cred, options);
+
+                return processor;
+
+            }, requiresCredential: false);
+    }
+
+    private BlobContainerClient GetBlobContainerClient(
+        AzureMessagingEventHubsSettings settings, TokenCredential cred, string configurationSectionName)
+    {
+        if (string.IsNullOrEmpty(settings.BlobClientConnectionName))
+        {
+            // throw an invalid operation exception if the blob client connection name is not provided
+            throw new InvalidOperationException(
+                $"A EventProcessorClient could not be configured. Ensure a valid connection name was provided in the '{configurationSectionName}' configuration section.");
+        }
+
+        var blobConnectionString =
+            builderConfiguration.GetConnectionString(
+                settings.BlobClientConnectionName) ??
+                throw new InvalidOperationException(
+                    "A EventProcessorClient could not be configured. " +
+                    $"There is no connection string in Configuration with the name {settings.BlobClientConnectionName}. " +
+                    "Are you sure you defined a Blob resource with this name in your AppHost?");
+
+        // FIXME: ideally this should be pulled from services; but thar be dragons.
+        // There is no reliable way to get the blob service client from the services collection
+        var blobUriBuilder = new BlobUriBuilder(new Uri(blobConnectionString!));
+
+        // consumer group and blob container names have similar constraints (alphanumeric, hyphen) but we should sanitize nonetheless
+        var suffix = (string.IsNullOrWhiteSpace(settings.ConsumerGroup)) ? "default" : settings.ConsumerGroup;
+        blobUriBuilder.BlobContainerName = $"checkpoints-{suffix}"; // TODO: should be unique to this consumer group
```

Comment:
I'd suggest changing the container name to something like `"{namespace}{eventhub}{consumergroup}`, as it will contain both checkpoints and ownership information for processors in the cluster.  The processor will create Blob Storage pseudo-folder thingies for "checkpoints" and "ownership" in the container.  The limitations on Blob container names make it hard to do this both readable and unique.

I don't think you should assume that you want to create a brand new unique container for every component.  You want the processors to be able to easily join a cluster for the same namespace/Event Hub/consumer group combination, as that is necessary for effective scaling.  The processor has an unbounded per-partition concurrency model and for most real-world scenarios, an application will need to have multiple processors sharing work to avoid overloading a single host.